Patches/Moyo light in the abyss/ThingDefs_Misc/Weapons.xml

```diff
@@ -0,0 +1,685 @@
+				</value>
+			</li>
+			
+			<li Class="PatchOperationAdd">
+				<xpath>Defs/ThingDef[defName="MeleeWeapon_PlasmaTachi"]</xpath>
+				<value>
+					<equippedStatOffsets>
+					  <MeleeCritChance>0.95</MeleeCritChance>
+					  <MeleeParryChance>0.54</MeleeParryChance>
+					  <MeleeDodgeChance>0.33</MeleeDodgeChance>
+					</equippedStatOffsets>
+				</value>
+			</li>
+			
+			<li Class="PatchOperationReplace">
+				<xpath>Defs/ThingDef[defName="MeleeWeapon_PlasmaTachi"]/tools</xpath>
+				<value>
+				  <tools>
+					  <li Class="CombatExtended.ToolCE">
+						<label>blade</label>
+						<capacities>
+						  <li>Cut</li>
+						</capacities>
+						<power>56</power>
+						<extraMeleeDamages>
+							<li>
+								<def>Burn</def>
+								<amount>7</amount>
+							</li>
+						</extraMeleeDamages>
+						<cooldownTime>2.1</cooldownTime>
+						<chanceFactor>1</chanceFactor>
+						<armorPenetrationBlunt>6.24</armorPenetrationBlunt>
+						<armorPenetrationSharp>16.26</armorPenetrationSharp>
+						<linkedBodyPartsGroup>Edge</linkedBodyPartsGroup>
+					  </li>
+					  <li Class="CombatExtended.ToolCE">
+						<label>blade</label>
+						<capacities>
+						  <li>Stab</li>
+						</capacities>
+						<power>14</power>
+						<extraMeleeDamages>
+							<li>
+								<def>Burn</def>
+								<amount>5</amount>
+							</li>
+						</extraMeleeDamages>
+						<cooldownTime>1.85</cooldownTime>
+						<chanceFactor>1</chanceFactor>
+						<armorPenetrationBlunt>3.46</armorPenetrationBlunt>
+						<armorPenetrationSharp>24</armorPenetrationSharp>
+						<linkedBodyPartsGroup>Point</linkedBodyPartsGroup>
+					  </li>
+					  <li Class="CombatExtended.ToolCE">
+						<label>handle</label>
+						<capacities>
+						  <li>Poke</li>
+						  <li>Blunt</li>
+						</capacities>
+						<power>7</power>
+						<cooldownTime>2.22</cooldownTime>
+						<chanceFactor>0.2</chanceFactor>
+						<armorPenetrationBlunt>2.4</armorPenetrationBlunt>
+						<linkedBodyPartsGroup>Handle</linkedBodyPartsGroup>
+					  </li>
+				  </tools>
+				</value>
+			</li>
+			
+			<!-- ========== Pistol =========== -->
+			
+			<li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+			<defName>RedMoyo_Pistol</defName>
+				<statBases>
+					<SightsEfficiency>1.00</SightsEfficiency>
+					<ShotSpread>0.17</ShotSpread>
+					<SwayFactor>0.57</SwayFactor>
+					<Bulk>4.50</Bulk>
+					<Mass>1.20</Mass>
+					<RangedWeapon_Cooldown>0.36</RangedWeapon_Cooldown>
+				</statBases>
+				
+				<Properties>
+					<verbClass>CombatExtended.Verb_ShootCE</verbClass>
+					<hasStandardCommand>true</hasStandardCommand>
+					<defaultProjectile>Bullet_MabyssalPlasmaPistol</defaultProjectile>
+					<warmupTime>0.6</warmupTime>
+					<range>16</range>
+					<soundCast>Shot_ChargeRifle</soundCast>
+					<soundCastTail>GunTail_Light</soundCastTail>
+					<muzzleFlashScale>9</muzzleFlashScale>
+				</Properties>
+				
+				<AmmoUser>
+					<magazineSize>35</magazineSize>

```

Comment:
Charge weapons usually benefit from more compact ammo, but this does seem a bit high for a pistol. Maybe around 20 or so would be more appropriate?

Reply:
Model is visually closer to a carbine or SMG really, and it's treated as two handed. That being said, just could be due to it being lower production quality than the actual military gear for a energy weapon, so I'll drop the magsize to 20.